### PR TITLE
chore: update product-links for modeler

### PR DESCRIPTION
**Pull request by barmac**
## Description

This updates `product-links.txt` to include all of the docs links which will be present in the upcoming Desktop Modeler release (8th August).

Some of the links already receive status code 301:

| Link | Status code |
| ---- | ------------ |
| https://docs.camunda.io/docs/components/modeler/bpmn/user-tasks/#assignments | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/exclusive-gateways/#conditions | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/business-rule-tasks/#defining-a-task | 200 |
| https://docs.camunda.io/docs/guides/utilizing-forms/#connect-your-form-to-a-bpmn-diagram | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/receive-tasks/#messages | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/message-events/#messages | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/call-activities/ | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/service-tasks/#task-definition | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/business-rule-tasks/#defining-a-task | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/script-tasks/#defining-a-task | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/send-tasks/#defining-a-task | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/message-events/#message-throw-events | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/multi-instance/#defining-the-collection-to-iterate-over | 200 |
| https://docs.camunda.io/docs/components/modeler/bpmn/error-events/#defining-the-error | 200 |
| https://docs.camunda.io/docs/components/concepts/variables/#input-mappings | 200 |
| https://docs.camunda.io/docs/components/concepts/variables/#output-mappings | 200 |
| https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-date | 301 |
| https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-cycle | 301 |
| https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-duration | 301 |
| https://docs.camunda.io/docs/components/modeler/feel/what-is-feel | 301 |
| https://docs.camunda.io/?utm_source=modeler&utm_medium=referral | 200 |
| https://docs.camunda.io/?utm_source=modeler&utm_medium=referral | 200 |
| https://docs.camunda.io/docs/guides/migrating-from-Camunda-Platform/ | 301 |
| https://docs.camunda.io/docs/components/modeler/desktop-modeler/model-your-first-diagram/ | 200 |
| https://docs.camunda.io/docs/components/modeler/desktop-modeler/?utm_source=modeler&utm_medium=referral | 301 |
| https://docs.camunda.io/docs/components/concepts/variables/?utm_source=modeler&utm_medium=referral | 200 |
| https://docs.camunda.io/docs/components/modeler/desktop-modeler/troubleshooting/#i-cannot-connect-to-zeebe | 200 |

The links were gathered via command: `grep -r -oh --extended-regexp "https://docs.camunda.io[^\"]+" .`

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

---

Related to https://github.com/camunda/developer-experience/issues/67

### CODE REVIEW

File: product-links.txt [hunk 1]
```diff
@@ -30,16 +30,36 @@ https://docs.camunda.io/docs/components/modeler/bpmn/user-tasks/user-tasks/
...
+https://docs.camunda.io/docs/components/concepts/variables/#output-mappings
+https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-date
+https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-cycle
+https://docs.camunda.io/docs/reference/bpmn-processes/timer-events/timer-events#time-duration
+
+# Desktop Modeler
+
+https://docs.camunda.io/docs/components/modeler/feel/what-is-feel
+https://docs.camunda.io/?utm_source=modeler&utm_medium=referral
+https://docs.camunda.io/?utm_source=modeler&utm_medium=referral
```
Comment (akeller) [inline product-links.txt:57]: Duplicates, but can be removed.

**Comment by akeller** [comment]
@christinaausley, I would only use this for Desktop Modeler & properties panel links. IMO, this could close https://github.com/camunda/developer-experience/issues/67 and https://github.com/camunda/camunda-platform-docs/issues/1680.

We can add a backlog item to fail this check on a redirect.

**Review by christinaausley — APPROVED** [review]
Thank you for opening this @barmac! Adjustments in latest commit to solely keep prop panel and Desktop Modeler links. Adding an issue to our backlog.

**Comment by barmac** [comment]
Thank you for the review!

**Pull request merged**